router/pkg/mcpserver/server.go

```diff
@@ -0,0 +1,676 @@
+		logger:                    options.Logger,
+		httpClient:                httpClient,
+		requestTimeout:            options.RequestTimeout,
+		routerGraphQLEndpoint:     routerGraphQLEndpoint,
+		excludeMutations:          options.ExcludeMutations,
+		enableArbitraryOperations: options.EnableArbitraryOperations,
+		exposeSchema:              options.ExposeSchema,
+		operationsManager:         operationsManager,
+		schemaCompiler:            schemaCompiler,
+	}
+
+	return gs, nil
+}
+
+// WithGraphName sets the graph name
+func WithGraphName(graphName string) func(*Options) {
+	return func(o *Options) {
+		o.GraphName = graphName
+	}
+}
+
+// WithOperationsDir sets the operations directory
+func WithOperationsDir(operationsDir string) func(*Options) {
+	return func(o *Options) {
+		o.OperationsDir = operationsDir
+	}
+}
+
+// WithServer sets the server configuration
+func WithServer(server ServerOptions) func(*Options) {
+	return func(o *Options) {
+		o.Server = server
+	}
+}
+
+// WithPort sets just the port
+func WithPort(port int) func(*Options) {
+	return func(o *Options) {
+		o.Server.Port = port
+	}
+}
+
+func WithRequestTimeout(timeout time.Duration) func(*Options) {
+	return func(o *Options) {
+		o.RequestTimeout = timeout
+	}
+}
+
+func WithLogger(logger *zap.Logger) func(*Options) {
+	return func(o *Options) {
+		o.Logger = logger
+	}
+}
+
+// WithExcludeMutations sets the exclude mutations option
+func WithExcludeMutations(excludeMutations bool) func(*Options) {
+	return func(o *Options) {
+		o.ExcludeMutations = excludeMutations
+	}
+}
+
+// WithEnableArbitraryOperations sets the enable arbitrary operations option
+func WithEnableArbitraryOperations(enableArbitraryOperations bool) func(*Options) {
+	return func(o *Options) {
+		o.EnableArbitraryOperations = enableArbitraryOperations
+	}
+}
+
+// WithExposeSchema sets the expose schema option
+func WithExposeSchema(exposeSchema bool) func(*Options) {
+	return func(o *Options) {
+		o.ExposeSchema = exposeSchema
+	}
+}
+
+// LoadOperations loads operations from the configured operations directory
+func (s *GraphQLSchemaServer) LoadOperations() error {
+	return s.operationsManager.LoadOperationsFromDirectory(s.operationsDir)
+}
+
+// LoadOperationsFromDirectory loads operations from a specified directory
+func (s *GraphQLSchemaServer) LoadOperationsFromDirectory(operationsDir string) error {
+	return s.operationsManager.LoadOperationsFromDirectory(operationsDir)
+}
+
+// ServeSSE starts the server with SSE transport
+func (s *GraphQLSchemaServer) ServeSSE() (*server.SSEServer, *http.Server, error) {
+	listenAddr := "localhost:" + strconv.Itoa(s.port)
+
+	// Create HTTP server with timeouts
+	httpServer := &http.Server{
+		Addr:         listenAddr,
+		ReadTimeout:  15 * time.Second,
+		WriteTimeout: 15 * time.Second,
+		IdleTimeout:  30 * time.Second,
+	}
+
+	sseServer := server.NewSSEServer(s.server,
+		server.WithBaseURL(fmt.Sprintf("http://%s", listenAddr)),
```

Comment:
what if the public URL is https?

Reply:
Not supported for now. It would mean someone runs the router with custom TLS which is rare because most of the users have a reverse proxy in front that does TLS offloading. We can support anytime.